server/src/main/java/io/unitycatalog/server/service/IcebergRestCatalogService.java

```diff
@@ -171,7 +171,7 @@ public LoadTableResponse loadTable(@Param("namespace") String namespace,
                                      @Param("table") String table) throws IOException {
     List<String> namespaceParts = Splitter.on(".").splitToList(namespace);
     if (namespaceParts.size() != 2) {
-      throw new IllegalArgumentException("invalid namespace " + namespace);
+      throw new BaseException(ErrorCode.INVALID_ARGUMENT, "invalid namespace " + namespace);
```

Comment:
Working on tests.

Also just realized, for this to be spec compliant we should make sure errors get surfaced with the IcebergErrorResponse https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml#L3197

Reply:
Thank you for catching this!

Reply:
Added tests, I've created a separate issue [here](https://github.com/unitycatalog/unitycatalog/issues/46) so we can track effort to make all the error responses spec compliant. Seemed easier for review if we tackle that separately since it touches more parts.